File: source/algorithm/NQ050/NQ050_improved.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int T;
    cin >> T;
    while (T--) {
        int n;
        cin >> n;
        vector<int> nums(n);
        // 读取n个数字
        for (int i = 0; i < n; i++) {
            cin >> nums[i];
        }
        sort(nums.begin(), nums.end()); // 对数组进行排序
        // 输出排序结果，注意最后一个元素后无空格
        for (int i = 0; i < n; i++) {
            if (i > 0) cout << " ";
            cout << nums[i];
        }
        cout << endl;
    }
    return 0;
}

File: 兴趣班入门百练讲义/NQ100/NQ050/NQ050_improved.py
import sys

def main():
    input = sys.stdin.read().split()
    ptr = 0
    T = int(input[ptr])
    ptr += 1
    
    for _ in range(T):
        n = int(input[ptr])
        ptr += 1
        nums = list(map(int, input[ptr:ptr+n]))  # 严格读取n个数字
        ptr += n
        nums.sort()  # 对数组进行排序
        print(' '.join(map(str, nums)))  # 输出排序结果，用空格连接

if __name__ == "__main__":
    main()
